tests/core/usage_statistics/test_usage_statistics_data_context.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import inspect
+import logging
+from unittest import mock
+
+import pytest
+
+from great_expectations.core.usage_statistics.events import UsageStatsEvents
+from great_expectations.core.usage_statistics.usage_statistics import ENABLED_METHODS
+
+logger = logging.getLogger(__name__)
+
+
+def filter_enabled_methods(
+    enabled_methods: dict[str, UsageStatsEvents]
+) -> dict[str, UsageStatsEvents]:
+    return {
+        k.split(".")[1]: v
+        for k, v in enabled_methods.items()
+        if k.startswith("AbstractDataContext") and not k.endswith("__init__")
+    }
+
+
+@pytest.fixture
+def enable_usage_stats(monkeypatch):
+    monkeypatch.delenv("GE_USAGE_STATS")
+
+
+@pytest.mark.unit
+def test_enabled_methods_map_to_appropriate_usage_stats_events():
+    assert filter_enabled_methods(ENABLED_METHODS) == {
+        "add_datasource": UsageStatsEvents.DATA_CONTEXT_ADD_DATASOURCE,
+        "build_data_docs": UsageStatsEvents.DATA_CONTEXT_BUILD_DATA_DOCS,
+        "get_batch_list": UsageStatsEvents.DATA_CONTEXT_GET_BATCH_LIST,
+        "open_data_docs": UsageStatsEvents.DATA_CONTEXT_OPEN_DATA_DOCS,
+        "run_checkpoint": UsageStatsEvents.DATA_CONTEXT_RUN_CHECKPOINT,
+        "run_profiler_on_data": UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_ON_DATA,
+        "run_profiler_with_dynamic_arguments": UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_WITH_DYNAMIC_ARGUMENTS,
+        "run_validation_operator": UsageStatsEvents.DATA_CONTEXT_RUN_VALIDATION_OPERATOR,
+        "save_expectation_suite": UsageStatsEvents.DATA_CONTEXT_SAVE_EXPECTATION_SUITE,
+    }
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@pytest.mark.parametrize(
+    "method_name,expected_event",
+    [(name, event) for name, event in filter_enabled_methods(ENABLED_METHODS).items()],
+)
+@pytest.mark.parametrize(
+    "data_context_fixture_name",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param("in_memory_runtime_context", id="EphemeralDataContext"),
```

Comment:
Do you want to ensure that if more `DataContext` subclasses get added, they are tested here? That is you could have a test that looks for all subclasses of `DataContext` and verify you only see these 3. 
This comment is not blocking just something to think about.